ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java

```diff
@@ -549,6 +570,96 @@ private static void createPhaseTapChanger(UcteAngleRegulation ucteAngleRegulatio
         ptca.add();
     }
 
+    private static void createRatioAndPhaseTapChanger(UcteAngleRegulation ucteAngleRegulation, UctePhaseRegulation uctePhaseRegulation, TwoWindingsTransformer transformer) {
+
+        LOGGER.trace("Create phase tap changer combining both ratio and phase tap '{}'", transformer.getId());
+
+        // When there is both Ratio and Phase tap changers the combined action of the two can be described with the formula as follows:
+        // neq(nr,np).dUeq.sin(ThetaEq) = np.dUp.(nr.dUr + 1).sin(Theta)
+        // neq(nr,np).dUeq.cos(ThetaEq) = nr.dUr + np.dUp.(nr.dUr + 1).cos(Theta)
+        // where np is the tap number of phase changer and nr is the tap number of the ratio changer
+        // this should lead to a double entry table where the equivalent neq tap number depends both of nr and np
+        // We propose the following approximation : we compute dUeq and ThetaEq at fixed current ratio tap nr = nro, we compute rho(nro)
+        // and build both equivalent phase tap changer and ratio tap changer such that it will be exact at ratio = nro
+
+        int lowerRatioTap = getLowTapPosition(uctePhaseRegulation, transformer);
+        RatioTapChangerAdder rtca = transformer.newRatioTapChanger()
+                .setLowTapPosition(lowerRatioTap)
+                .setTapPosition(uctePhaseRegulation.getNp())
+                .setLoadTapChangingCapabilities(!Double.isNaN(uctePhaseRegulation.getU()));
+        if (!Double.isNaN(uctePhaseRegulation.getU())) {
+            rtca.setLoadTapChangingCapabilities(true)
+                    .setRegulating(true)
```

Comment:
I am trying to refactor (and understand) this PR. In case of combination, I understand that the approximation is right only for ratio tap changer current position. So for me we maybe have to set regulating boolean to false.

Reply:
Agreed